src/App.js

```diff
@@ -29,11 +29,27 @@ const App = () => {
   // This starts state off as a 2D array of JS objects with
   // empty value and unique ids.
   const [squares, setSquares] = useState(generateSquares());
+  const [currentPlayer, setCurrentPlayer] = useState(PLAYER_1);
 
   // Wave 2
-  // You will need to create a method to change the square
-  //   When it is clicked on.
-  //   Then pass it into the squares as a callback
+  const onClickCallback = (updatedSquare) => {
+    // You will need to create a method to change the square
+    //   When it is clicked on.
+    //   Then pass it into the squares as a callback
+    const newSquares = squares.map((square) => {
```

Comment:
This section of code isn't working because squares is a 2D array.  In this case there may be a way to get `map` to work (perhaps by nesting) but I recommend a nested for loop.